### DISCUSSION TITLE
Memory Card game rules

### DISCUSSION THREAD

**Discussion by pd-code** [TheOdinProject/curriculum#25361] [General]
In the [memory card game project](https://www.theodinproject.com/lessons/javascript-memory-card), [this solution](https://github.com/heldersrvio/memory-card-game) is given as a reference. I ran into the special case of getting 12 answers right, but then only these 12 cards were offered for the next round, effectively ending the game. I checked the code, and there is no guard against this case. Shouldn't there be at least one live card per round until the end, so you can theoretically reach a maximum score of 14 in every game?

**Comment by KevinMulhern** [comment]
@TheOdinProject/javascript anybody have any thoughts on this?

**Comment by thatblindgeye** [comment]
It might be worth including an instruction that you should make sure at least one new card is displayed when updated... but really that's up to the user if they want to include that/think to include it. The main purpose of the project is to utilize hooks/lifecycle methods and state.

- We could either include that as an optional criterion
- We remove the reference to that specific project. I don't think we really do that much anywhere else so it doesn't make sense to do it here. If we want to provide a reference then we could try linking to an external site, or just update the instructions to be clearer.

**Reply by wise-king-sullyman** [reply]
I think your second option here would probably be the best way to address this, as considering the goals of the project having it in the spec (optional or not) wouldn't really benefit learners IMO, as you've also highlighted.
